legend-depot-core-services/src/main/java/org/finos/legend/depot/services/projects/DependenciesCache.java

```diff
@@ -77,15 +78,28 @@ private Function3<String,String,String,StoreProjectVersionData> getProjectDataFr
 
     private Set<ProjectVersion> calculateTransitiveDependencies(ProjectVersion projectVersion, Map<String, StoreProjectVersionData> projectDataMap)
     {
-        return getTransitiveDependencies(projectVersion, getProjectDataFromProjectsMap(projectDataMap));
+        return getTransitiveDependencies(projectVersion, getProjectDataFromProjectsMap(projectDataMap), false);
     }
 
-    private Set<ProjectVersion> getTransitiveDependencies(ProjectVersion pv, Function3<String,String,String,StoreProjectVersionData> projectDataProvider)
+    private Set<ProjectVersion> getTransitiveDependencies(ProjectVersion pv, Function3<String,String,String,StoreProjectVersionData> projectDataProvider, boolean checkDep)
     {
         Set<ProjectVersion> dependencies = new HashSet<>();
         try
         {
-            StoreProjectVersionData projectData = projectDataProvider.value(pv.getGroupId(), pv.getArtifactId(), pv.getVersionId());
+            StoreProjectVersionData projectData = null;
+            if (checkDep)
```

Comment:
i dont think we need the code below here, the project data provider should really checking a version exist,
so if the result is null and we have the flag to check deps we throw an exception

this method getProjectDataFromStore() should already fail if the version is not there .get() would fail

wjhat happens when we initialise the cache, how are checking we are not building with mssing deps in stiore